ansible_base/api_documentation/preprocessing_hooks.py

```diff
@@ -0,0 +1,157 @@
+    return view.__class__
+
+
+def _extract_prefix_from_path(path):
+    """Extract the operation_id prefix from a URL path."""
+    path_parts = parse_path_segments(path)
+    if not path_parts:
+        return None, None, None
+
+    prefix = path_parts[-1]
+    path_parts_count = len(path_parts)
+    return prefix, path_parts, path_parts_count
+
+
+def _create_compound_prefix(path_parts, fallback_prefix):
+    """Create compound prefix from path parts (e.g., 'orgs_teams') to resolve collisions."""
+    if len(path_parts) >= 2:
+        return '_'.join(path_parts[-2:])
+    return fallback_prefix
+
+
+def _handle_prefix_collision(prefix, class_name, path_parts_count, path_parts, operation_class_map):
+    """
+    Handle collision when multiple ViewSets use the same operation_id prefix.
+    ViewSet with fewer path parts gets simple prefix; the other gets compound prefix.
+    """
+    existing_class, existing_count, existing_path_parts = operation_class_map[prefix]
+
+    # Same ViewSet class - no collision
+    if existing_class == class_name:
+        return prefix
+
+    # Different ViewSet - resolve collision
+    if path_parts_count < existing_count:
+        # Current is main resource - move existing to compound prefix
+        # Use the existing entry's path_parts to create the correct compound prefix
+        compound_prefix = _create_compound_prefix(existing_path_parts, prefix)
+        operation_class_map[compound_prefix] = (existing_class, existing_count, existing_path_parts)
+        operation_class_map[prefix] = (class_name, path_parts_count, path_parts)
+        logger.debug(f"Resource collision: {class_name} (main, {path_parts_count} parts) owns '{prefix}', {existing_class} moved to '{compound_prefix}'")
+        return prefix
+    else:
+        # Existing is main resource - current gets compound prefix
+        compound_prefix = _create_compound_prefix(path_parts, prefix)
+        operation_class_map[compound_prefix] = (class_name, path_parts_count, path_parts)
+        logger.debug(f"Resource collision: {existing_class} (main, {existing_count} parts) keeps '{prefix}', {class_name} stored at '{compound_prefix}'")
+        return compound_prefix
+
+
+def _register_skip_ai_description(view_class, class_name, prefix):
+    """Register a ViewSet that should skip AI description generation."""
+    if getattr(view_class, 'skip_ai_description', False):
+        SKIP_AI_DESCRIPTION_PREFIXES.add(prefix)
+        logger.info(f"View class {class_name} (prefix: {prefix}) has skip_ai_description=True")
+
+
+def _register_resource_purpose(view_class, class_name, prefix):
+    """Register a ViewSet's resource_purpose for description generation."""
+    resource_purpose = getattr(view_class, 'resource_purpose', None)
+    if resource_purpose:
+        RESOURCE_PURPOSE_MAP[class_name] = resource_purpose
+        logger.debug(f"View class {class_name} (prefix: {prefix}) has resource_purpose: {resource_purpose[:50]}...")
```

Comment:
This would print out the ... even if the resoure_purpose is < 50 characters. I don't really care, just pointing it out.

Reply:
Valid. And updated.